st2common/st2common/util/schema/__init__.py

```diff
@@ -164,33 +169,49 @@ def modify_schema_allow_default_none(schema):
     defines a default value of None.
     """
     schema = copy.deepcopy(schema)
-
     properties = schema.get('properties', {})
+
     for property_name, property_data in six.iteritems(properties):
+        is_optional = not property_data.get('required', False)
         has_default_value = 'default' in property_data
         default_value = property_data.get('default', None)
 
-        if has_default_value and default_value is None:
-            # Allow "None" to be also used as a valid attribute value
-            property_type = schema['properties'][property_name]['type']
-
-            if isinstance(property_type, list) and 'null' not in property_type:
-                schema['properties'][property_name]['type'].append('null')
-            elif isinstance(property_type, six.string_types) and 'null' not in property_type:
-                schema['properties'][property_name]['type'] = [property_type, 'null']
+        if (has_default_value or is_optional) and default_value is None:
+            # If property is anyOf and oneOf then it has to be process differently.
+            if set(['anyOf', 'oneOf']) & set(schema['properties'][property_name].keys()):
```

Comment:
This is a bit hard to read and follow - would it be possible to split in a separate method or similar?